src/js/view/controls/components/menu/menu.js

```diff
@@ -0,0 +1,499 @@
+            this.items = [];
+            emptyElement(this.el);
+            menuItems.forEach(menuItem => {
+                this.items.push(menuItem);
+                this.el.appendChild(menuItem.el);
+            });
+            if (initialSelectionIndex > -1) {
+                menuItems[initialSelectionIndex].activate();
+            }
+            this.categoryButton.show();
+        }
+    }
+    appendMenu(childMenu) {
+        if (!childMenu) {
+            return;
+        }
+        // eslint-disable-next-line no-shadow
+        const { el, name, categoryButton } = childMenu;
+        this.children[name] = childMenu;
+        if (categoryButton) {
+            const topbar = this.mainMenu.topbar;
+            const sharingButton = topbar.querySelector('.jw-settings-sharing');
+            const appendBefore = name === 'quality' ? topbar.firstChild : sharingButton || this.closeButton.element();
+
+            topbar.insertBefore(
+                categoryButton.element(),
+                appendBefore
+            );
+        } else if (this.topbar) {
+            const submenuItem = new MenuItem(name, childMenu.open);
+            this.topbar.appendChild(submenuItem.el);
+        } else {
+            this.createTopBar();
+        }
+        this.el.appendChild(el);
+    }
+    removeMenu(name) {
+        if (!name) {
+            return this.parentMenu.removeMenu(this.name);
+        }
+        
+        if (!this.children[name]) {
+            return;
+        }
+        const menu = this.children[name];
+        delete this.children[name];
+        menu.destroy();
+    }
+    open(evt) {
+        // if visible, return
+        if (this.visible) {
+            return;
+        }
+        if (this.isSubmenu) {
+            if (this.mainMenu.openMenus.length) {
+                this.mainMenu.closeAll();
+            }
+            if (this.items) {
+                this.items[0].el.focus();
+            }
+            this.el.classList.add('jw-settings-submenu-active');
+            this.categoryButton.element().setAttribute('aria-checked', 'true');
+            this.mainMenu.openMenus.push(this.name);
+            if (!this.mainMenu.visible) {
+                this.mainMenu.open(evt);
+                if (this.items && evt.type === 'enter') {
+                    this.items[0].el.focus();
+                } else {
+                    // Don't show tooltip if auto-focusing for navigation's sake.
+                    this.categoryButton.tooltip.suppress = true;
+                    this.categoryButton.element().focus();
+                }
+            }
+        } else {
+            this.el.parentNode.classList.add('jw-settings-open');
+            this.trigger('menuVisibility', { visible: true, evt });
+            document.addEventListener('click', this.onDocumentClick);
+        }
+        this.visible = true;
+        this.el.setAttribute('aria-expanded', 'true');        
+    }
+    close(evt) {
+        // if visible, return
```

Comment:
conditional is correct, comment is not

Reply:
bump this comment

Reply:
Yep, removing these unnecessary comments.